pkg/storage/replica.go

```diff
@@ -3301,6 +3308,12 @@ func (r *Replica) processRaftCommand(
 			"command %s proposed from replica %+v: %s",
 			raftCmd.BatchRequest, raftCmd.OriginReplica, err,
 		)
+		// TODO(spencer,andrei): This error is confusing if r.mu.state.Lease is
```

Comment:
I think the thing to do here is to return `LeaseRejectedError`, and I think we'll need to augment the code in `redirectOnOrAcquire` that interprets that error and currently converts it into a `notLeaseHolderError` to maybe recognize that the error might mean that in fact we the lease that we wanted might have been applied (in case the `LeaseRejectedError` is coming from a re-proposal).